catalog/plugin/catalog-plugin-metadatabackup/src/main/java/org/codice/ddf/catalog/plugin/metadata/backup/MetadataBackupPlugin.java

```diff
@@ -0,0 +1,313 @@
+        if (metacardTransformerProperty instanceof String && StringUtils.isNotBlank((String) metacardTransformerProperty)) {
+            setMetacardTransformerId((String) metacardTransformerProperty);
+            LOGGER.debug("Updating {} with {}",
+                    METACARD_TRANSFORMER_ID_PROPERTY, metacardTransformerProperty);
+        }
+
+        Object keepDeletedMetacards = properties.get(KEEP_DELETED_METACARDS_PROPERTY);
+        if (keepDeletedMetacards instanceof Boolean) {
+            this.keepDeletedMetacards = (Boolean) keepDeletedMetacards;
+            LOGGER.debug("Updating {} with {}", KEEP_DELETED_METACARDS_PROPERTY,
+                    keepDeletedMetacards);
+        }
+
+        Object folderDepth = properties.get(FOLDER_DEPTH_PROPERTY);
+        if (folderDepth instanceof Integer) {
+            setFolderDepth((Integer) folderDepth);
+            LOGGER.debug("Updating {} with {}", FOLDER_DEPTH_PROPERTY, folderDepth);
+        }
+    }
+
+    private void processRequest(ProcessRequest<? extends ProcessResourceItem> processRequest)
+            throws PluginExecutionException {
+        LOGGER.trace("Backing up metadata");
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new PluginExecutionException(
+                    "Unable to backup ingested metadata; no outputDirectory.");
+        }
+
+        if (metacardTransformer == null) {
+            throw new PluginExecutionException(
+                    "Unable to backup ingested metadata; no Metacard Transformer found.");
+        }
+
+        List<? extends ProcessResourceItem> processResourceItems = processRequest.getProcessItems();
+        processResourceItems.forEach(processResourceItem -> {
+            Metacard metacard = processResourceItem.getMetacard();
+            try {
+                LOGGER.trace("Backing up metadata for metacard : {}", metacard.getId());
+                BinaryContent binaryContent = metacardTransformer.transform(metacard, null);
+                copyMetadataToOutputDirectory(binaryContent, metacard.getId());
+            } catch (CatalogTransformerException e) {
+                LOGGER.debug("Unable to transform metacard with id {}.", metacard.getId(), e);
+            }
+        });
+    }
+
+    private void copyMetadataToOutputDirectory(BinaryContent content, String metacardId) {
+        if (content == null || content.getInputStream() == null) {
+            LOGGER.debug("No content for transformed metacard {}.", metacardId);
+            return;
+        }
+
+        File parent = getCompleteDirectory(getFolderDepth(), metacardId);
+        if (!parent.exists()) {
+            try {
+                FileUtils.forceMkdir(parent);
+            } catch (IOException e) {
+                LOGGER.debug("Unable to make directory {}", parent.getPath(), e);
+            }
+        }
+
+        validateDirectory(parent);
+        File file = new File(parent.getPath() + File.separator + metacardId);
```

Comment:
You should really be using https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html and related classes for all of these operations to avoid running into file system issues.